src/Symfony/Component/Console/Helper/Helper.php

```diff
@@ -118,7 +119,7 @@ public static function formatMemory(int $memory)
 
     public static function strlenWithoutDecoration(OutputFormatterInterface $formatter, $string)
     {
-        return self::strlen(self::removeDecoration($formatter, $string));
+        return (new UnicodeString(self::removeDecoration($formatter, $string)))->width();
```

Comment:
I'd suggest patching the strlen method instead, and to use `UnicodeString` only `if (preg_match('//u', $string))`